tiled/catalog/adapter.py

```diff
@@ -1040,6 +1041,34 @@ async def write_block(self, *args, **kwargs):
             (await self.get_adapter()).write_block, *args, **kwargs
         )
 
+    async def patch(self, *args, **kwargs):
+        # assumes a single DataSource (currently only supporting zarr)
+        async with self.context.session() as db:
+            new_shape_and_chunks = await ensure_awaitable(
+                (await self.get_adapter()).patch, *args, **kwargs
+            )
+            node = await db.get(orm.Node, self.node.id)
+            data_source = node.data_sources[0]
```

Comment:
I know that currently only one DataSource is supported, but should we add a raise if there is more than one is present, mostly to remind whomever tries to implement multiple DataSource?